src/representation_theory/basic.lean

```diff
@@ -0,0 +1,113 @@
+/-
+Copyright (c) 2022 Antoine Labelle. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Labelle
+-/
+import algebra.module.basic
+import data.equiv.module
+import algebra.module.linear_map
+import algebra.monoid_algebra.basic
+import linear_algebra.trace
+
+/-!
+# Monoid representations
+
+This file introduces monoid representations and their characters and proves basic lemmas about them,
+including equivalences between different definitions of representations.
+
+## Main definitions
+
+  * `as_module`
+  * `rep_space`
+  * `character`
```

Comment:
Can you please add the full name here? I mean `namespace.name` (in doing so we will have a working hyperlink).